storage/cloud.py

```diff
@@ -1032,6 +1034,14 @@ def get_direct_download_url(
                 path, request_ip, expires_in, requires_cors, head, **kwargs
             )
 
+        if kwargs:
+            namespace = kwargs.get("namespace")
+            if namespace is not None:
+                if namespace in self.cloudfront_distribution_org_overrides:
+                    self.cloudfront_distribution_domain = (
```

Comment:
you are updating a class variable here. What happens if there are other requests at the same time which may depend on this variable and it may wrongly be set. There's only one object of `CloudFrontedS3Storage` so all requests will use that same variable. A better approach here would be to use a local variable for the domain